libs/parallelism/algorithms/include/hpx/parallel/algorithms/inclusive_scan.hpp

```diff
@@ -499,32 +398,496 @@ namespace hpx { namespace parallel { inline namespace v1 {
+                hpx::execution::seq, first, last, dest,
+                std::plus<value_type>());
+        }
+
+        // clang-format off
+        template <typename ExPolicy, typename FwdIter1, typename FwdIter2,
+            HPX_CONCEPT_REQUIRES_(
+                hpx::is_execution_policy<ExPolicy>::value &&
+                hpx::traits::is_forward_iterator<FwdIter1>::value &&
+                hpx::traits::is_forward_iterator<FwdIter2>::value
+            )>
+        // clang-format on
+        friend typename parallel::util::detail::algorithm_result<ExPolicy,
+            FwdIter2>::type
+        tag_fallback_dispatch(hpx::inclusive_scan_t, ExPolicy&& policy,
+            FwdIter1 first, FwdIter1 last, FwdIter2 dest)
+        {
+            static_assert(hpx::traits::is_forward_iterator<FwdIter1>::value,
+                "Requires at least forward iterator.");
+            static_assert(hpx::traits::is_forward_iterator<FwdIter2>::value,
+                "Requires at least forward iterator.");
+
+            using value_type =
+                typename std::iterator_traits<FwdIter1>::value_type;
+
+            return hpx::parallel::v1::detail::inclusive_scan<FwdIter2>().call(
+                std::forward<ExPolicy>(policy), first, last, dest,
+                std::plus<value_type>());
+        }
+
+        // clang-format off
+        template <typename InIter, typename OutIter, typename Op,
+            HPX_CONCEPT_REQUIRES_(
+                hpx::traits::is_iterator<InIter>::value &&
+                hpx::traits::is_iterator<OutIter>::value &&
+                hpx::is_invocable_v<Op,
+                    typename std::iterator_traits<InIter>::value_type,
+                    typename std::iterator_traits<InIter>::value_type
+                >
+            )>
+        // clang-format on
+        friend OutIter tag_fallback_dispatch(hpx::inclusive_scan_t,
+            InIter first, InIter last, OutIter dest, Op&& op)
+        {
+            static_assert((hpx::traits::is_input_iterator<InIter>::value),
+                "Requires at least input iterator.");
+            static_assert((hpx::traits::is_output_iterator<OutIter>::value),
+                "Requires at least output iterator.");
+
+            return hpx::parallel::v1::detail::inclusive_scan<OutIter>().call(
+                hpx::execution::seq, first, last, dest, std::forward<Op>(op));
+        }
+
+        // clang-format off
+        template <typename ExPolicy, typename FwdIter1, typename FwdIter2,
+            typename Op,
+            HPX_CONCEPT_REQUIRES_(
+                hpx::is_execution_policy<ExPolicy>::value &&
+                hpx::traits::is_forward_iterator<FwdIter1>::value &&
+                hpx::traits::is_forward_iterator<FwdIter2>::value &&
+                hpx::is_invocable_v<Op,
+                    typename std::iterator_traits<FwdIter1>::value_type,
+                    typename std::iterator_traits<FwdIter1>::value_type
+                >
```

Comment:
The concept checks for the operation have not been added consistently everywhere (see also `exclusive_scan`). Could you please add those?